ethers-solc/src/lib.rs

```diff
@@ -287,6 +289,48 @@ impl<T: ArtifactOutput> Project<T> {
         project::ProjectCompiler::with_sources(self, Source::read_all(files)?)?.compile()
     }
 
+    /// Convenience function to compile only (re)compile files that match the provided [FileFilter].
+    /// Same as [`Self::svm_compile()`] but with only with those files as input that match
+    /// [FileFilter::is_match()].
+    ///
+    /// # Example - Only compile Test files
+    ///
+    /// ```
+    /// use ethers_solc::{Project, TestFileFilter};
+    /// # fn demo(project: Project) {
+    /// let project = Project::builder().build().unwrap();
+    /// let output = project
+    ///     .compile_sparse(
+    ///         TestFileFilter::default()
+    ///     ).unwrap();
+    /// # }
+    /// ```
+    ///
+    /// # Example - Apply a custom filter
+    ///
+    /// ```
+    /// use std::path::Path;
+    /// use ethers_solc::Project;
+    /// # fn demo(project: Project) {
+    /// let project = Project::builder().build().unwrap();
+    /// let output = project
+    ///     .compile_sparse(
+    ///         |path: &Path| path.ends_with("Greeter.sol")
+    ///     ).unwrap();
+    /// # }
+    /// ```
+    #[cfg(all(feature = "svm", feature = "async"))]
+    pub fn compile_sparse<F: FileFilter + 'static>(
```

Comment:
Instead of `compile_sparse`, should we add a filter in the Project struct / in the builder? and then `compile()` would check if there's a filter and override the sources with sparse output?

Reply:
this would work, but has the disadvantage that the filter is then tied to the project so you'd need to modify the project if you need another filter, on recompile with a different filter for example.
I think this also won't add convenience to foundry because what we do is `config.project()`, if we add a filter to the project itself it will look like `config.project().with_filter(..).compile()` instead of `config.project().compile_sparse(..)`